### ISSUE TITLE
next() yields a time in the past

### ISSUE THREAD

**Issue by rvdwenden** [alonsodomin/cron4s#59]
When executing the next() function on a cron with weekdays we sometimes get None or a localdatetime in the past (same for prev() ...):
```scala
import java.time._

import cron4s.lib.javatime._
import cron4s._

object Cron4Spec {
  def main(args: Array[String]): Unit = {
    val Right(cron) = Cron("0 0 0 * * 1-3")

    for (dayOfMonth <- 1 to 30) {
      val from = LocalDateTime.of(2017, 3, dayOfMonth, 2, 0, 0)
      cron.next(from) match {
        case Some(next) =>
          println(s" $from: after ${next.isAfter(from)}")
        case None =>
          println(s"Cannot get next from: $from")
      }

      cron.prev(from) match {
        case Some(prev) =>
          println(s" $from: before ${prev.isBefore(from)}")
        case None =>
          println(s"Cannot get prev from: $from")
      }
    }
  }
}
```

**Comment by rvdwenden** [comment]
produced with version 0.3.1.

**Comment by alonsodomin** [comment]
Thanks for reporting, I'll look at that as that is a weird behaviour

**Comment by alonsodomin** [comment]
This has been fixed in branch 0.3.x and version 0.3.2 released to Maven Central (it may take a while until is mirrorred everywhere).

**Comment by rvdwenden** [comment]
Works for us now after fix (0.3.2)!
Thanks for fixing.

**Comment by alonsodomin** [comment]
Thanks for confirming. Beware that #56 is still lurking around but quite to fix in 0.3.x as I need to change a bit the internals. I'm sort of close of a cut for 0.4.0 though which will have that fixed.
